blinkit-app/src/homePage/organisms/HomePage.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import vegetables, { total } from "../../DATA/ITEMS";
+import { UserProvider } from "../../userContext";
+import Foodbar from "../molecules/foodbar/Foodbar";
+import Footer from "../molecules/footer/Footer";
+import Header from "../molecules/header/Header";
+import MainPage from "../molecules/mainPage/MainPage";
+import Speciality from "../molecules/specialities/Speciality";
+import Information from "../molecules/information/Information"
+
+class HomePage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: vegetables,
+      total: total,
+      query: "",
+      queryOutput: vegetables,
+    };
+  }
+  add = (index) => {
+    console.log(index);
+    vegetables[index].quantity = vegetables[index].quantity + 1;
+    total.item += 1;
+    total.price += Number(vegetables[index].currentPrice);
+    this.setState({
+      items: vegetables,
+      total: total,
+    });
+  };
+  minus = (index) => {
```

Comment:
Why do we need this function in the component? This can be a part of a separate helper file and then we can import it here.

As on every mount this will be computed, which we need to avoid.
Second, DRY(Do not repeat yourself)
If this a helper, it can be reused at other places.
Third, It willl not pollute the Component file, which increases readability